vite.js

```diff
@@ -4,91 +4,112 @@ const processCSS = require('./src/process-css');
 const babelPlugin = require('./babel.js');
 const NAME = require('./package.json').name;
 
-const VIRTUAL_MODULE_NAME = '@virtual-style9';
+const VIRTUAL_MODULE_NAME = '\0plugin-style9:virtual-module';
+const VIRTUAL_CSS_NAME = VIRTUAL_MODULE_NAME + '.css';
 
+async function transformStyle9(
+  code,
+  id,
+  { parserOptions, restOptions, server, cssModules }
+) {
+  const res = await babel.transformAsync(code, {
+    plugins: [[babelPlugin, restOptions]],
+    filename: id,
+    sourceMaps: true,
+    parserOpts: parserOptions,
+    babelrc: false
+  });
+  if (!res) {
+    return;
+  }
+  const { code: _code, map, metadata } = res;
+  const output = [_code];
+  const css = metadata.style9;
+  if (css) {
+    cssModules.set(id, css);
+    output.unshift(`import '${VIRTUAL_MODULE_NAME}'`);
+    if (server) {
+      const { moduleGraph } = server;
+      const virtualModule = moduleGraph.getModuleById(VIRTUAL_CSS_NAME);
+      if (virtualModule) {
+        const { code } = virtualModule.transformResult;
+        const output = code.split('\n');
+        const finalCoe = output.reduce((acc, cur) => {
+          if (cur.startsWith('const __vite__css')) {
+            cur = `const __vite__css = '${genCSS(cssModules)}'`;
+          }
+          return (acc += cur + '\n');
+        }, '');
+        virtualModule.transformResult.code = finalCoe;
+      }
+    }
+  }
+  return {
+    map,
+    code: output.join('\n')
+  };
+}
+function genCSS(cssModule) {
+  const chunk = Array.from(cssModule.values()).reduce(
+    (acc, cur) => (acc += cur),
+    ''
+  );
```

Comment:
Is this not just a more complicated `.join('')`?

Reply:
You're right. Code has been simplified :)